File: python3/495.py
class Solution:
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        size = len(timeSeries)
        ans = duration * size;
        for i in range(1, len(timeSeries)):
            ans -= max(0, duration-(timeSeries[i]-timeSeries[i-1]))
        return ans

File: c++11/495.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int size = timeSeries.size();
        int ans = duration * size;
        for (int i=1; i<size; ++i) {
            ans -= max(0, duration-(timeSeries[i] - timeSeries[i-1]));
        }
        return ans;
    }
};
